src/coreclr/jit/gentree.cpp

```diff
@@ -7541,6 +7541,8 @@ GenTree* Compiler::gtNewStructVal(ClassLayout* layout, GenTree* addr)
         blkNode = gtNewObjNode(layout, addr);
     }
 
+    blkNode->SetIndirExceptionFlags(this);
```

Comment:
I am not too familiar with this, so could you elaborate how `SetIndirExceptionFlags` with the original problem?

Reply:
`SetIndirExceptionFlags` can only be called on trees that are indirs but this function can fold indirs of locals to `GT_LCL_VAR` nodes. So this change is just moving it into `gtNewStructVal` in the indir cases, similar to how it works for `gtNewIndir` today.

Reply:
Got it.